tsdb/head.go

```diff
@@ -2035,3 +2068,21 @@ func (ss stringset) slice() []string {
 	sort.Strings(slice)
 	return slice
 }
+
+// SeriesLifecycleCallback specifies a list of callbacks that will be called during a lifecycle of a series.
+// It is always a no-op in Prometheus and mainly meant for external users who import TSDB.
+type SeriesLifecycleCallback interface {
+	// PreCreationCallback is called before creating a series to indicate if the series can be created.
+	// A non nil error means the series should not be created.
+	PreCreationCallback(labels.Labels) error
```

Comment:
Maybe just without callback? we are already in callback interface. 
```suggestion
	PreCreation(labels.Labels) error
```
And below